lib/apiservers/engine/backends/container_portlayer.go

```diff
@@ -0,0 +1,1195 @@
+		container.MacAddress = ""
+		container.ExposedPorts = nil  //FIXME:  Add once port mapping is implemented
+		container.PublishService = "" // Name of the network service exposed by the container
+	}
+
+	// Get the original container config from the image's metadata in our image cache.
+	var imageConfig *metadata.ImageConfig
+
+	if info.ContainerConfig.LayerID != nil {
+		imageConfig, _ = cache.ImageCache().GetImage(*info.ContainerConfig.LayerID)
+	}
+
+	// Fill in the values with defaults from the original image's container config
+	// structure
+	if imageConfig != nil {
+		container.StopSignal = imageConfig.ContainerConfig.StopSignal // Signal to stop a container
+
+		container.OnBuild = imageConfig.ContainerConfig.OnBuild // ONBUILD metadata that were defined on the image Dockerfile
+
+		// Fill in information about the container's volumes
+		// FIXME:  Why does types.ContainerJSON have Mounts and also ContainerConfig,
+		// which also has Volumes?  Assuming this is a copy from image's container
+		// config till we figure this out.
+		container.Volumes = imageConfig.ContainerConfig.Volumes
+	}
+
+	return container
+}
+
+func networkFromContainerInfo(id string, info *models.ContainerInfo) *types.NetworkSettings {
+	if info == nil || info.ScopeConfig == nil {
+		return nil
+	}
+
+	networks := &types.NetworkSettings{
+		NetworkSettingsBase: types.NetworkSettingsBase{
+			Bridge:                 "",
+			SandboxID:              "",
+			HairpinMode:            false,
+			LinkLocalIPv6Address:   "",
+			LinkLocalIPv6PrefixLen: 0,
+			Ports:                  nil, //FIXME:  Fill in once port mapping is implemented.
+			SandboxKey:             "",
+			SecondaryIPAddresses:   nil,
+			SecondaryIPv6Addresses: nil,
+		},
+		Networks: nil,
+	}
+
+	return networks
+}
+
+func plContainerListToDockerContainerList(plContainerList []models.ContainerListInfo) []*types.Container {
+	containers := make([]*types.Container, 0, len(plContainerList))
+	for _, t := range plContainerList {
+		cmd := strings.Join(t.ExecArgs, " ")
+		// the docker client expects the friendly name to be prefixed
+		// with a forward slash -- create a new slice and add here
+		names := make([]string, 0, len(t.Names))
+		for i := range t.Names {
+			names = append(names, clientFriendlyContainerName(t.Names[i]))
+		}
+		c := &types.Container{
+			ID:      *t.ContainerID,
+			Image:   *t.RepoName,
+			Created: *t.Created,
+			Status:  *t.Status,
+			Names:   names,
+			Command: cmd,
+			SizeRw:  *t.StorageSize,
+		}
+		containers = append(containers, c)
+	}
+
+	return containers
+}
+
+// helper function to format the container name
+// to the docker client approved format
+func clientFriendlyContainerName(name string) string {
+	return fmt.Sprintf("/%s", name)
+}
+
+//----------
+// Utility Functions
+//----------
+
+func createNewAttachClientWithTimeouts(connectTimeout, responseTimeout, responseHeaderTimeout time.Duration) (*client.PortLayer, *httpclient.Transport) {
```

Comment:
Please name the parameters the same as the struct member names (Connect, ResponseHeader, Re*quest*).  Also please mirror the order assigned.